mkraft/node/asnoleader.go

```diff
@@ -166,7 +170,14 @@ func (n *Node) RunAsCandidate(ctx context.Context) {
 				case <-electionTicker.C:
 					// last election reaches no decisive result, either no response or not enough votes
 					// we re-elect
-					consensusChan = n.candidateAsyncDoElection(ctx)
+					electionCancel()
+					electionCtx, electionCancel = context.WithCancel(ctx)
```

Comment:
First, as with [line 129](https://github.com/maki3cat/mkraft/pull/50/files#diff-92d11675b6bb8b8b107cf14e901764a573d572d82d14a1b97e6c3b1aa42cc623R129), the election timeout here may be unnecessary. Then, why here uses `WithCancel`, but `WithTimeout` in the beginning?